modules/mbedtls/Kconfig

```
@@ -111,6 +112,37 @@ config MBEDTLS_DEBUG_LEVEL
 	  This makes Zephyr call mbedtls_debug_set_threshold() function during
 	  mbedTLS initialization, with the configured debug log level.
 
+choice MBEDTLS_DEBUG_EXTRACT_BASENAME
+	prompt "Extract basename from filenames"
+	default MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_BUILDTIME
+
+config MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_BUILDTIME
+	bool "At buildtime"
+	help
+	  Adds compile options, which should convert full source paths in
+	  __FILE__ macro to files' basenames. This will reduce code footprint
+	  when debug messages are enabled.
+
+	  This is compiler dependent, so if it does not work then please
+	  fallback to MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_RUNTIME instead.
+
+config MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_RUNTIME
+	bool "At runtime"
```

Just `runtime`, looks cleaner in menuconfig as selected choice in the
 `Extract basename from filenames (runtime)`
```suggestion
	bool "runtime"
```